core/validation/validate.py

```diff
@@ -169,6 +170,38 @@ def validate_uniques(
     return non_unique_failures
 
 
+def validate_unique_composite_keys(
+    workbook: dict[str, pd.DataFrame], sheet_name: str, composite_keys: tuple
+) -> list[vf.NonUniqueCompositeKeyFailure]:
+    """
+    Validates the uniqueness of composite keys across multiple columns.
+
+    :param workbook: A dictionary containing sheet names as keys and corresponding pandas DataFrames as values.
+    :param sheet_name: The name of the worksheet to be validated.
+    :param composite_keys: A list of tuples, where each tuple contains the column names
+                           that should have combined uniqueness.
+    :return: A list of non-unique composite key failures, if any exist.
+    """
+
+    sheet = workbook[sheet_name]
+    non_unique_composite_key_failures = []
+
+    # removing duplicates as to avoid index mismatch
+    sheet.drop_duplicates(subset=list(composite_keys), inplace=True)
+    # filter dataframe by these columns and find duplicated rows including NaN/Empty
+    mask = sheet[list(composite_keys)].duplicated(keep=False)
+    mask.drop_duplicates(inplace=True)
+
+    if mask.any():
+        failures = [
+            vf.NonUniqueCompositeKeyFailure(sheet=sheet_name, row=tuple(row[1:]))
+            for row in sheet[mask].itertuples(index=False)
+        ]
```

Comment:
you've implemented the suggestion of using `.itertuples(index=False)` but left in the row slicing `row[1:]` and therefore omitting the first legitimate value from the offending row

Reply:
also don't need to cast to a tuple - as the itertuples returns a type of tuple anyway